py/port_dml.py

```diff
@@ -1050,19 +1050,23 @@ def main(argv):
                 sys.stderr.write("%s error%s: %s\n" % (
                     loc, '' if e.tag is None else ' ' + e.tag, e))
                 sys.stderr.write("%s:%d: found here\n" % (tagfilename, lineno))
-                fail = True
+                errors += 1
             except:
                 sys.stderr.write("Unexpected error on this porting tag:\n")
                 sys.stderr.write(line)
                 sys.stderr.write("%s:%d: found here\n" % (tagfilename, lineno))
                 traceback.print_exc()
-                fail = True
+                errors += 1
 
     PWUNUSED.report(src, args.dest)
 
     with (open(args.dest, 'w', newline='') if args.dest else sys.stdout) as f:
         src.commit(f)
-    if fail:
+    if errors:
+        total = sum(map(len, transformations.values()))
+        sys.stderr.write(f'''\
+*** Failed to apply {errors} out of {total} porting tags; partial result saved to {args.dest}. Consider applying the failed tags manually.
```

Comment:
Great, how would one apply these tags manually?

Reply:
After injecting a bug in the porting script, the failure is presented as follows:
```
Unexpected error on this porting tag:
/disk1/ecarsten/6/src/tools/dmlc/test/1.2/misc/porting.dml:35:23: porting PHARD_RESET_VALUE: []
/disk1/ecarsten/6/logs/test/t126_dmlc/scratch/porting/tags:52: found here
Traceback (most recent call last):
  File "/disk1/ecarsten/6/linux64/bin/dml/python/port_dml.py", line 1048, in main
    t.apply(src)
  File "/disk1/ecarsten/6/linux64/bin/dml/python/port_dml.py", line 334, in apply
    assert act_before == exp_before, act_before
AssertionError: hard_reset_value
```
Looking at porting.dml:35, I see:
```
            parameter hard_reset_value = 3;
```
Looking at the same place in the generated file (now line 37), I see:
```
            param hard_reset_value = 3;
```
Now, if I search the manual for PHARD_RESET_VALUE, I find:

> The `hard_reset_value` parameter is no longer
> recognized. The parameter is automatically renamed to
> `init_val`, which has roughly the same effect.

so I can change line 37 to:
```
            param init_val = 3;
```

Reply:
Ok, I was just wondering how the wording `applying the failed tags` come into play here, can I correlate the errors to the tags file and see what needs to be ported?

Reply:
yes, please re-read my comment. If the final message says there are N errors, then it is preceded by N tracebacks on the mentioned form, the location in the tag file is mentioned, and that line is also displayed (`tags:52` reads `/disk1/ecarsten/6/src/tools/dmlc/test/1.2/misc/porting.dml:35:23: porting PHARD_RESET_VALUE: []`)

Reply:
Ah, got it. Sorry for the noise.